flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetReaders.java

```diff
@@ -116,7 +116,11 @@ public ParquetValueReader<RowData> struct(Types.StructType expected, GroupType s
         int id = field.fieldId();
         if (idToConstant.containsKey(id)) {
           // containsKey is used because the constant may be null
-          reorderedFields.add(ParquetValueReaders.constant(idToConstant.get(id)));
+
+          // We use the max definition level of the parent node to infer the max definition level of the constant field
+          // in case of we could not find the given parquet field with typesById.
+          int fieldD = type.getMaxDefinitionLevel(currentPath());
```

Comment:
nit: potential typo? `fieldD` reads oddly to me. Is that supposed to be fieldDepth or fieldId? I've been thinking it was `fieldId`, but now I'm not sure.

If it is `fieldId`, my follow up question is whether or not that's the most appropriate name for it given that it's an inferred max definition level.

If it's something else and `D` is an acronym, could you please name the field something more specific?